plugins/Actions/Actions.php

```diff
@@ -32,11 +33,35 @@ public function registerEvents()
             'AssetManager.getJavaScriptFiles' => 'getJsFiles',
             'Insights.addReportToOverview'    => 'addReportToInsightsOverview',
             'Metrics.getDefaultMetricTranslations' => 'addMetricTranslations',
+            'Metrics.getDefaultMetricSemanticTypes' => 'addMetricSemanticTypes',
             'Metrics.getDefaultMetricDocumentationTranslations' => 'addMetricDocumentationTranslations',
             'Actions.addActionTypes' => 'addActionTypes'
         );
     }
 
+    public function addMetricSemanticTypes(&$types)
+    {
+        $metrics = array(
+            'nb_pageviews'        => Metric::SEMANTIC_TYPE_NUMBER,
+            'nb_uniq_pageviews'   => Metric::SEMANTIC_TYPE_NUMBER,
+            'nb_downloads'        => Metric::SEMANTIC_TYPE_NUMBER,
+            'nb_uniq_downloads'   => Metric::SEMANTIC_TYPE_NUMBER,
+            'nb_outlinks'         => Metric::SEMANTIC_TYPE_NUMBER,
+            'nb_uniq_outlinks'    => Metric::SEMANTIC_TYPE_NUMBER,
+            'nb_searches'         => Metric::SEMANTIC_TYPE_NUMBER,
+            'nb_keywords'         => Metric::SEMANTIC_TYPE_NUMBER,
+            'avg_time_generation' => Metric::SEMANTIC_TYPE_DURATION, // TODO: should all have the same unit
```

Comment:
Any reason for adding this here and not directly in the metric class? Wouldn't it be useful to use the metric classes to define the semantic type where possible?
At some point in the future I would actually love to have metric classes for all available metrics and that reports need to use a metric class for each metric they contain. But that might be a bit bigger refactoring, so not sure when we will get there. But already moving everything to the Metric class that can be defined there might already be helpful....

Reply:
Most reports reference metrics by their string names, not by a Metric instance, so it would be difficult in Report.php to get that information. There is a getSemanticType() method in Metric though, but right now it's primarily used for ProcessedMetrics, which are currently used in Report::$processedMetrics.

Reply:
Sure. But isn't `avg_time_generation` a processed metric as well? But might be not that import for now. Just wanted to mention that we should try to move as much as possible to the metrics, so we have less work when we start refactoring that at some point...

Reply:
I just put the same metrics from the `'Metrics.getDefaultMetricTranslations'` hook into the new semantic types hook. I don't expect that would hinder the move from string metric names to metric classes in Report classes in any way. The work should take hours at most regardless.